app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt

```diff
@@ -500,6 +506,44 @@ class ReaderPresenter(
         }
     }
 
+    private fun downloadNextChapters() {
+        val manga = manga ?: return
+        val chaptersNumberToDownload = preferences.autoDownloadAfterReading().get()
+        if (chaptersNumberToDownload == 0 || !manga.favorite) return
+        val isChapterDownloaded = downloadManager.isChapterDownloaded(getCurrentChapter()!!.chapter, manga)
+        if (isChapterDownloaded) {
+            downloadAutoNextChapters(chaptersNumberToDownload)
```

Comment:
pass the chapter to this since we'll work backwards if only the previous chapter is downloaded
```suggestion
            val currentChapters = viewerChaptersRelay.value ?: return
            downloadAutoNextChapters(chaptersNumberToDownload, if (isChapterDownloaded) currentChapters.currChapter else currentChapters.prevChapter)
```